### DISCUSSION TITLE
如何使用微信云开发

### DISCUSSION THREAD

**Discussion by MrHeer** [weapp-vite/weapp-vite#55] [Q&A]
我通过 `npx weapp-vite create my-app` 创建的项目，请问如何使用[微信云开发](https://developers.weixin.qq.com/miniprogram/dev/wxcloudservice/wxcloud/basis/getting-started.html)

> 我目前通过微信云官方文档，在 `project.config.json` 中配置了 `cloudfunctionRoot`，经过测试可以调用云函数。请问有什么办法可以使用 `TypeScript` 编写云函数吗？

**Comment by sonofmagic** [comment]
这个实际上也是可以的，因为内置了各种 typescript 的打包工具，可以进行处理，具体可以在这个 issue 里面讨论具体的形式，比如 cloudfunctionRoot 下的一个一个函数，都暴露一个 index.ts, 然后整体按照目录打包到 cloudfunctionRoot_dist 这样的方式
你有什么好的建议和意见可以在这里提。

**Reply by MrHeer** [reply]
如果有办法自动生成云函数的调用端函数将会非常不错。现在小程序调用云函数也没有类型，并且还要手动输入云函数名称。

**Reply by MrHeer** [reply]
不知道能不能装饰器来做这件事，如果可以应该很不错。我可以在任何地方定义云函数

```ts
@cloudFn
function sum(a: number, b: number) {
  return a + b;
}
```
